cadl-tests/cadl/apiversion.cadl

```diff
@@ -0,0 +1,29 @@
+import "@cadl-lang/rest";
+
+using Cadl.Http;
+
+@service({
+  title: "ApiVersion",
+  version: "1.0.0",
+})
+@server(
+  "{endpoint}/{ApiVersion}",
+  "Api Version",
+  {
+    endpoint: string,
+
+    @doc("Api Version")
+    @path
+    ApiVersion: APIVersions,
+  }
+)
+namespace Cadl.ApiVersion;
+
+enum APIVersions {
+  v1_0: "v1.0",
+}
+
+@route("/apiversion")
+interface ApiVersionOp {
+  read(): string;
```

Comment:
What does this operation do? Should we can add a query parameter named `api-version`?

Reply:
We tested api version in path param. https://github.com/Azure/autorest.java/blob/c3d52e44fd85c31c835e3b176828b3117ca16a47/cadl-tests/src/main/java/com/cadl/apiversion/implementation/ApiVersionClientImpl.java#L140